eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -131,19 +131,21 @@ extends:
           - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
             parameters:
               Paths:
-              - '/*'
-              - '!sdk/**/test-recordings/*'
-              - '!sdk/**/recordings/*'
-              - '!sdk/**/SessionRecords/*'
-              - '!sdk/**/session-records/*'
+                - '/*'
+                - '!sdk/**/test-recordings/*'
+                - '!sdk/**/recordings/*'
+                - '!sdk/**/SessionRecords/*'
+                - '!sdk/**/session-records/*'
               Repositories:
-              - Name: $(SpecRepoOwner)/$(SpecRepoName)
-                Commitish: $(SpecRepoCommit)
-                WorkingDirectory: $(SpecRepoDirectory)
-              - Name: $(SdkRepoOwner)/$(SdkRepoName)
-                Commitish: ${{ parameters.SdkRepoCommit }}
-                WorkingDirectory: $(SdkRepoDirectory)
+                - Name: $(SpecRepoOwner)/$(SpecRepoName)
+                  Commitish: $(SpecRepoCommit)
+                  WorkingDirectory: $(SpecRepoDirectory)
+                - Name: $(SdkRepoOwner)/$(SdkRepoName)
+                  Commitish: ${{ parameters.SdkRepoCommit }}
+                  WorkingDirectory: $(SdkRepoDirectory)
               SkipCheckoutNone: true
+              ${{ if endsWith(parameters.SpecRepoUrl, '-pr') }}:
```

Comment:
I think we should add a condition that this pipeline is running in internal as well because otherwise the secret will not be available.